.github/workflows/tests.yaml

```diff
@@ -36,10 +37,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python "3.11"
+      - name: Set up Python "3.12"
```

Comment:
In bd81e52, I've updated the code to run mypy on all runtimes supported by mypy since we should make sure static analysis passes on all versions. All checks passed on all runtimes